gnss_lib_py/parsers/android.py

```diff
@@ -422,278 +562,169 @@ def preprocess(self, input_path):
+                    continue
+                if row[0][0] == '#':    # header row
+                    if len(row) == 1:
+                        continue
+                    sensor_field = row[0][2:]
+                    if sensor_field in self.sensor_fields:
+                        sensor_data[sensor_field] = [row[1:]]
                 else:
-                    if row[0] == 'Accel':
-                        accel.append(row[1:])
-                    elif row[0] == 'Gyro':
-                        gyro.append(row[1:])
-
-        accel = pd.DataFrame(accel[1:], columns = accel[0],
-                             dtype=np.float64)
-        gyro = pd.DataFrame(gyro[1:], columns = gyro[0],
-                            dtype=np.float64)
-
-        #Drop common columns from gyro and keep values from accel
-        gyro.drop(columns=['utcTimeMillis', 'elapsedRealtimeNanos'],
-                  inplace=True)
-        measurements = pd.concat([accel, gyro], axis=1)
-        #NOTE: Assuming pandas index corresponds to measurements order
-        #NOTE: Override times of gyro measurements with corresponding
-        # accel times
-        return measurements
+                    if row[0] in self.sensor_fields:
+                        sensor_data[row[0]].append(row[1:])
 
-    @staticmethod
-    def _row_map():
-        row_map = {'AccelXMps2' : 'acc_x_mps2',
-                   'AccelYMps2' : 'acc_y_mps2',
-                   'AccelZMps2' : 'acc_z_mps2',
-                   'GyroXRadPerSec' : 'ang_vel_x_radps',
-                   'GyroYRadPerSec' : 'ang_vel_y_radps',
-                   'GyroZRadPerSec' : 'ang_vel_z_radps',
-                   }
-        return row_map
+        sensor_dfs = [pd.DataFrame(data[1:], columns = data[0],
+                                   dtype=np.float64) for _,data in sensor_data.items()]
 
+        # remove empty dataframes
+        sensor_dfs = [df for df in sensor_dfs if len(df) > 0]
 
-class AndroidRawFixes(NavData):
-    """Class handling location fix measurements from raw Android dataset.
+        if len(sensor_dfs) == 0:
+            measurements = pd.DataFrame()
+        elif len(sensor_dfs) > 1:
+            measurements = pd.concat(sensor_dfs, axis=1)
+        else:
+            measurements = sensor_dfs[0]
 
-    Inherits from NavData().
-    """
-    def __init__(self, input_path):
-        pd_df = self.preprocess(input_path)
-        super().__init__(pandas_df=pd_df)
+        return measurements
 
+    def postprocess(self):
+        """Postprocess loaded data."""
 
-    def preprocess(self, input_path):
-        """Read Android raw file and produce location fix dataframe objects
+        # add gps milliseconds
+        self["gps_millis"] = unix_to_gps_millis(self["unix_millis"])
 
-        Parameters
-        ----------
-        input_path : string or path-like
-            File location of data file to read.
+    def _row_map(self):
+        """Map of column names from loaded to gnss_lib_py standard
```

Comment:
Similarly, here the docstring has `column` but elsewhere variables are labelled as `row`.

Reply:
updated 10 places in the repo